src/DataFrame/IO/Parquet/Page.hs

```diff
@@ -127,13 +127,16 @@ readPageTypeHeader hdr@(DictionaryPageHeader{..}) xs lastFieldId =
                             identifier
                 3 ->
                     let
-                        (isSorted : rem') = rem
+                        (isSorted, rem') = case rem of
```

Comment:
Nit: this repetition looks a little strange. I think the canonical way to do this would be using uncons but that means we should bump up our version of base since the function was added recently. Maybe:

```haskell
let
    isSorted = fromMaybe (error ...) (listToMaybe rem)
in ... (drop 1 rem) ...
```

Feels slightly better.